### PR TITLE
Build a Ruby 3.2.3/bookworm image

### PR DESCRIPTION
**Pull request by sulami**
The base images in question are
- https://hub.docker.com/layers/library/ruby/3.2.3-bookworm/images/sha256-5590025acebb13cacd5a5a4e5b0733c18d841318b8b788996bb88ce1fc64c565
- https://hub.docker.com/layers/library/ruby/3.2.3-slim-bookworm/images/sha256-7bd0053d5820b233c060775c8fe21d25a9f3ee1ae4fb04bc8fe7887f9c44a2f3

This should get us onto OpenSSL 3, which is the point of this, as well as pick up a slightly newer Ruby version.

### CODE REVIEW

**Comment by sulami** [comment]
I've used this image to run a hats review app and confirmed that we don't have the GPayments issue anymore, my payment went through just fine.

image

File: .github/workflows/build-rails-base.yml [hunk 1]
```diff
@@ -27,8 +30,8 @@ jobs:
     - name: prepare
       run: |-
         apk add --no-cache python3 py3-pip
-        pip3 install --upgrade pip
-        pip3 install awscli
+        pip3 install --break-system-packages --upgrade pip
+        pip3 install --break-system-packages awscli
```
Comment (showwin) [inline .github/workflows/build-rails-base.yml:34]: Did you face any problems in this step for the 3.2.3 image?

File: .github/workflows/build-rails-base.yml [hunk 2]
```diff
@@ -14,6 +14,9 @@ jobs:
           - ruby: '3.2.2'
             folder: '3.1.x' # slim bullseye
             tag: '3.2.2-slim@sha256:81743b016046dd7e6fe55b3b408ac37735786c9dff409ca2191daffa0a66fa7e'
+          - ruby: '3.2.3'
+            folder: '3.1.x' # slim bookworm
```
Comment (showwin) [inline .github/workflows/build-rails-base.yml:18]: If we use `3.1.x/Dockerfile` for other versions, should we rename the folder as `3.x.x`...?

**Review by showwin — APPROVED** [review]
Left some comments for my curiosity.

LGTM 👍

File: .github/workflows/build-rails-base.yml (hunk 1 above)
Comment (sulami) [reply .github/workflows/build-rails-base.yml:34]: Yes, see [`ccf5de2` (#21)](https://github.com/degica/dockerfiles/pull/21/commits/ccf5de27d5fd305f6cfbf4bb8ac0990c3245cf1a).

File: .github/workflows/build-rails-base.yml (hunk 2 above)
Comment (sulami) [reply .github/workflows/build-rails-base.yml:18]: I thought about that, but I didn't want to add more change than strictly necessary here, and we're already using it for other 3.2.x images.

File: .github/workflows/build-rails-base.yml (hunk 1 above)
Comment (showwin) [reply .github/workflows/build-rails-base.yml:34]: Ah, thanks for your explanation in detail!

**Pull request merged**